source/extensions/filters/http/jwt_authn/verifier.cc

```diff
@@ -297,10 +299,20 @@ class AnyVerifierImpl : public BaseGroupVerifierImpl {
     if (completion_state.is_completed_) {
       return;
     }
+    // For RequireAny: usually it returns the error from the last provider.
+    // But if a Jwt is not for a provider, its auth returns JwtMissed or JwtUnknownIssuer.
+    // Such error should not be used as the final error if there are other valid errors.
+    if (status != Status::Ok && status != Status::JwtMissed && status != Status::JwtUnknownIssuer) {
```

Comment:
It makes sense why we check `JwtUnknownIssuer`, we should use the status from the provider the JWT is intended for. 

But why check for `JwtMissed`? Doesn't this mean the request is missing a token? If that is the case, wouldn't all providers return this exact same error?

Reply:
Actually, it makes sense for JwtMissed if you understand how Extractor works.  An extractor uses jwt_location from its provider. If a jwt is in location a,  but provider specifies location b, it could not extract any tokens, it will return JwtMissed.

In some complidated case "all_fail" or "all_fail_and_missing",  extractor has all jwt_locations from all providers.  Extractor can extract some tokens,  but when passed to a provider for authentication, the provider will double check its issuer, and found it is not intended for it, it will return JwtUnknowIssuer. 

That is why we have to special handle two errors: JwtMissed, and JwtUnknownIssuer.